### PR TITLE
Implemented basic recording functionality

### PR DESCRIPTION
**Pull request by TimGels**
Added the recording functionality to the dashcam fragment kotlin and
xml.
Added drawables for use in the dashcam fragment ui.
Added colors for use in the dashcam fragment ui.
Added strings for use in the dashcam fragment ui.

Added required dependencies to the gradle app file.

It will save recorded videos with a timestamp in the filename to prevent
overwriting old files. The saved video files can be accessed with ADB in
`/data/data/com.camerasecuritysystem.client/files`.

### CODE REVIEW

**Comment by TimGels** [comment]
> 1. The pause and resume button functionality, both in code and xml, seems still there, while the camera can only start and stop recording
> 
>     2. Every time I open the camera, it shows a toast with "You have not accepted all the permissions" even though I have accepted the camera permission
> 
>     3. Is it supposed to show the file path after saving? It doesn't do that here

1 - 2) good catch, i will take a look into it.

3) It is not supposed to show the path to the user, the user can't access the files without ADB or this app own functionality (or root) so a path is not useful.

**Comment by TimGels** [comment]
1 and 2 should be fixed now.

**Comment by JohannesKauffmann** [comment]
There are still XML files for the pause and resume UI elements.

**Comment by JohannesKauffmann** [comment]
2 is also not fixed here, both from a reïnstall and subsequent runs.

**Comment by TimGels** [comment]
> 2 is also not fixed here, both from a reïnstall and subsequent runs.

I cannot reproduce this myself.

**Comment by TimGels** [comment]
> There are still XML files for the pause and resume UI elements.

should be removed now

**Comment by JohannesKauffmann** [comment]
Well, now the app doesn't ask for permission at all anymore...

**Review by JohannesKauffmann — APPROVED** [review]

File: app/src/main/res/layout/fragment_dashcam.xml [hunk 1]
```diff
@@ -1,19 +1,49 @@
...
 
-
     <androidx.camera.view.PreviewView
-        android:id="@+id/viewFinder"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
+        android:id="@+id/preview_view"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:background="@color/bkPurple"
```
Comment (JochemFB) [inline app/src/main/res/layout/fragment_dashcam.xml:15]:
```suggestion
        android:background="@color/black"
```

**Review by JochemFB — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_dashcam.xml (hunk 1 above)
Comment (TimGels) [reply app/src/main/res/layout/fragment_dashcam.xml:15]: I will also remove the color from colors.xml

**Review by JochemFB — APPROVED** [review]

**Pull request merged**